Feature/endorse teammates

nuzurie opened the pull request.

nuzurie committed:
STUD-120/domain, Tag: create tag endpoint, usecase and repository

nuzurie committed:
Refactor: fix formatting

nuzurie committed:
STUD-109/Review: add addReview feature

nuzurie committed:
STUD-110/Review: allow reviews to be updated

nuzurie committed:
STUD-111/review, student: add reviews for student

nuzurie committed:
STUD-112/domain, review: return reviews made by a student

nuzurie committed:
STUD-121/app: plug auth middleware

nuzurie committed:
STUD-121: reduce technical debt & fix tests

Some places had hard coded values b/c these values were supposed to be extracted form the auth service. Now that the auth service is connected, everything works as intended

nuzurie committed:
STUD-121: add circuit breaker

aminebechrifa commented on Review/delivery/http/review_delivery.go:
```diff
@@ -0,0 +1,142 @@
+		c.JSON(http.StatusUnauthorized, errors.NewUnauthorizedError("not allowed to get reviews by others"))
+	}
+
+	var student domain.Student
```
to be removed , not used later

aminebechrifa commented on Review/delivery/http/review_delivery.go:
```diff
@@ -0,0 +1,142 @@
+	ctx := c.Request.Context()
+
+	var student domain.Student
+	err = c.ShouldBindJSON(&student)
```
not used later

nuzurie committed:
Refactor: remove the bugs with binding student for no reason

aminebechrifa commented on Review/delivery/http/review_delivery.go:
```diff
@@ -0,0 +1,123 @@
+
+	loggedID, _ := c.Get("loggedID")
+	logged, _ := loggedID.(string)
+	if logged != reviewer {
```
return after this

aminebechrifa commented on Review/delivery/http/review_delivery.go:
```diff
@@ -0,0 +1,123 @@
+		c.JSON(http.StatusUnauthorized, errors.NewUnauthorizedError("not allowed to get reviews by others"))
+	}
+
+	if loggedID != reviewer {
```
return after this

nuzurie committed:
Refactor: address lint issues and PR review comments

marksman1298 committed:
Test/school classes (#5)

* STUD-118 creating mock School_repository

* STUD-118/ Added TestSearchSchoolByDomain

added a small modification to the paramether in school repository

* STUD-118/ Updated TestSearchByDomain

* Added School Use Case Mock class

* STUD-118/ Fixed TestSearchByDomain

* Added tests for sendConfirmation and ConfirmSchoolEnrollment

* WIP: Test Search Student School

* Added tests for school delivery

* STUD-118/ TestSendConfirmationMail adced inSchool_delivery_test.go

* STUD-118/TestSendConfirmationMail final version School_delivery_test.go

* Small changes to school delivery tests

* STUD-118 creating mock School_repository

* STUD-118/ Added TestSearchSchoolByDomain

added a small modification to the paramether in school repository

* STUD-118/ Updated TestSearchByDomain

* Added School Use Case Mock class

* Added tests for sendConfirmation and ConfirmSchoolEnrollment

* WIP: Test Search Student School

* STUD-118/ Fixed TestSearchByDomain

* STUD-118/ TestSendConfirmationMail adced inSchool_delivery_test.go

* Added tests for school delivery

* STUD-118/TestSendConfirmationMail final version School_delivery_test.go

* Small changes to school delivery tests

* Fixed url for school delivery tests

Co-authored-by: Adam Yafout <adam.yafout@gmail.com>
Co-authored-by: Zubair Nurie <45270163+nuzurie@users.noreply.github.com>

aminebechrifa committed:
Test/endorse teammates (#7)

* testing tags and reviews

* Update review_usecase.go

* Update tag_repository.go

* Update tag_usecase.go

nuzurie commented on Review/delivery/http/review_delivery.go:
```diff
@@ -0,0 +1,131 @@
+	}
+
+	loggedID, _ := c.Get("loggedID")
+	reviewer, _ := loggedID.(string)
```
Bug: make sure the loggedID is the same as the reviewer's ID

nuzurie commented on Review/delivery/http/review_delivery.go:
```diff
@@ -0,0 +1,131 @@
+	ctx := c.Request.Context()
+
+	loggedID, _ := c.Get("loggedID")
+	reviewer, _ := loggedID.(string)
```
set the reviewer's ID in the review

nuzurie commented on Review/delivery/http/review_delivery.go:
```diff
@@ -0,0 +1,53 @@
+
+	var review domain.Review
+	err := c.ShouldBindJSON(&review)
+	if err != nil || review.Reviewed.ID == "" || review.Tags == nil {
```
Make this conditional inline

SanaJallouli committed:
Style the email for confirmation

car12lin12 reviewed (approved):
Nice job
